feat(docs-infra): Adds copy link to anchor functionality

SkyZeroZx opened the pull request:
Enables copying a direct link to any section by clicking its anchor. Also updates the aria-label to remove the code tag.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
Currently when clicked in clipboard icon only navigate to anchor and don't copy o anything 


## What is the new behavior?
Add a tooltip that provides a similar user experience to the one used in the Chrome developer docs.

  
<img width="468" height="89" alt="image" src="https://github.com/user-attachments/assets/b50ea356-c381-46fe-8f4a-0cc0d6eba1cd" />

Also normalized text when `code` tag is used in aria label

<img width="637" height="161" alt="image" src="https://github.com/user-attachments/assets/68bcb602-955a-45c0-870f-257e62dee3ed" />

<img width="599" height="73" alt="image" src="https://github.com/user-attachments/assets/f559cc30-94c7-415e-a196-88b5214ca515" />



## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

JeanMeche commented:
Thanks for the suggestion, I like the general idea but the tooltip is a bit too much visual noise imho. 

Maybe we should only copy the lick (and show the tooltip) when hovering that link icon ?

JeanMeche commented on adev/shared-docs/pipeline/shared/marked/transformations/heading.mts:
```diff
@@ -41,10 +41,11 @@ export function formatHeading(
   const customId = customIdRegex.exec(anchorLessText)?.[1];
   const link = customId ?? getHeaderId(anchorLessText);
   const label = anchorLessText.replace(/`(.*?)`/g, '<code>$1</code>').replace(customIdRegex, '');
+  const normalizedLabel = label.replace(/<\/?code>/g, '');
```
I see improvement that could add, the icon should become visible on heading hover, not only the icon itself.

JeanMeche commented on adev/shared-docs/components/viewers/docs-viewer/docs-viewer.component.spec.ts:
```diff
@@ -208,4 +216,41 @@ describe('DocViewer', () => {
+
+    const copyButton = fixture.nativeElement.querySelector(
+      'docs-copy-link-button',
+    ) as HTMLButtonElement;
```
```suggestion
    );
```
nit

SkyZeroZx replied:
Done

JeanMeche commented on adev/shared-docs/components/viewers/docs-viewer/docs-viewer.component.spec.ts:
```diff
@@ -208,4 +216,41 @@ describe('DocViewer', () => {
+
+    const copyButton = fixture.nativeElement.querySelector(
+      'docs-copy-link-button',
+    ) as HTMLButtonElement;
```
```suggestion
    );
```
Type assertion looks unnecesary

JeanMeche commented on adev/shared-docs/components/viewers/docs-viewer/docs-viewer.component.spec.ts:
```diff
@@ -208,4 +216,41 @@ describe('DocViewer', () => {
+    const fixture = TestBed.createComponent(DocViewer);
+    fixture.componentRef.setInput('docContent', exampleContentWithDocsAnchor);
+
+    fixture.detectChanges();
```
Our tests are zoneless, usually we shouldn't need to call detectChanges

JeanMeche reviewed (commented):
I also noticed that we can improve the vertical alignment on the icon

SkyZeroZx committed:
feat(docs-infra): Adds copy link to anchor functionality

Enables copying a direct link to any section by clicking its anchor. Also updates the aria-label to remove the code tag

SkyZeroZx commented on adev/shared-docs/styles/_anchor.scss:
```diff
@@ -1,18 +1,4 @@
 @mixin docs-anchor() {
```
Done
